.github/workflows/pr_python.yaml

```diff
@@ -0,0 +1,34 @@
+name: pr_python
+
+on:
+  push:
+    paths:
+      - "framework/python/**"
+      - "samples/python/**"
+  pull_request:
+    paths-ignore:
+      - "framework/python/**"
+      - "samples/python/**"
+    branches:
+      - main
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.PAT || secrets.GITHUB_TOKEN }}
+          
+      - name: MegaLinter Python flavor
+        uses: oxsecurity/megalinter/flavors/python@v6.12.0
+        env:
+          IGNORE_GITIGNORED_FILES: true
+          VALIDATE_ALL_CODEBASE: true
+          PRINT_ALL_FILES: true
+          DISABLE: SPELL,COPYPASTE,YAML
+          DISABLE_LINTERS: PYTHON_MYPY,REPOSITORY_CHECKOV,REPOSITORY_TRIVY
```

Comment:
PYTHON_PYRIGHT, and PYTHON_MYPY are both disabled in `.mega-linter.yml` though it wouldn't hurt to add PYTHON_PYRIGHT here since we aren't using static typing